### PR TITLE
Blaze: Update mappers for target options and remove stubbed response

### PR DESCRIPTION
**Pull request by itsmeichigo**
Part of #11622 

## Description

This PR updates the mappers for Blaze target options following the recent changes in the API response [peeHDf-2fw-p2#comment-1780]. The stubbed responses for the target endpoints were also removed since the endpoints have been deployed [peeHDf-2fw-p2#comment-1804]. 

An issue with the target device endpoint: it returns an "All" option. I added an update to the device picker view to remove the All section, and posted a question for the backend team [peeHDf-2fw-p2#comment-1808]. I'll update the view later if needed, depending on their response.

## Testing instructions

- Log in to a store with at least one existing product and is eligible for Blaze.
- Navigate to the Products tab and select any public product.
- On the product form, select Promote with Blaze.
- Notice that the campaign creation form is presented.
- Select the language field, note that the real data is now displayed.
- Navigate back and select the device field; the real data should be displayed on this screen, too. The All field is an option from the API, so picking it does not auto select all the other options for now.

## Screenshots

Language picker | Device picker 
--- | ---
 | 

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-ios/milestone/180">16.9</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-ios/milestone/180">16.9</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/Classes/ViewRelated/ReusableViews/SwiftUI Components/MultiSelectionList.swift [hunk 1]
```diff
@@ -20,7 +20,7 @@ struct MultiSelectionList<T: Hashable & Identifiable>: View {
         comment: "Placeholder in the search bar of a multiple selection list"
     )
 
-    init(allOptionsTitle: String,
+    init(allOptionsTitle: String? = nil,
```
Comment (hafizrahman) [inline WooCommerce/Classes/ViewRelated/ReusableViews/SwiftUI Components/MultiSelectionList.swift:23]: nice decision to make this optionally shown.

**Review by hafizrahman — APPROVED** [review]
At this point everything looks and works correctly to me.

I hope we get clarification about the devices situation soon because this is weird UX at the moment:

**Pull request merged**